### ISSUE TITLE
E0002 parsing I3013 on AWS::RDS::DBInstance if Engine is a Ref

### ISSUE THREAD

**Issue by drAlberT** [aws-cloudformation/cfn-lint#2385]
### CloudFormation Lint Version

0.65.0

### What operating system are you using?

Ubuntu 22.04

### Describe the bug

A cfn-lint exception is raised when parsing I3013 rule.

The trigger seems to be the presence of a reference as a value of the "Engine" resource parameter.

![Screenshot from 2022-09-29 13-57-09](https://user-images.githubusercontent.com/1442700/193029918-32a6b143-06c4-4707-834b-b9ea0c262c74.png)
![Screenshot from 2022-09-29 13-58-10](https://user-images.githubusercontent.com/1442700/193029945-c8d06db4-4513-4389-89eb-70ea6f51296e.png)
![Screenshot from 2022-09-29 14-15-37](https://user-images.githubusercontent.com/1442700/193029965-83f97802-cc82-459a-ac09-d424430ccb4e.png)

### Expected behavior

No error should be present is `Engine: !Ref Something` is used.

### Reproduction template

```yaml
---
AWSTemplateFormatVersion: 2010-09-09

Parameters:

  Engine:
    Description: DB Engine
    Type: String
    AllowedValues:
      - aurora-mysql
      - aurora-postgresql


Resources:

  DbCluster:
    Type: AWS::RDS::DBCluster
    DeletionPolicy: Snapshot
    UpdateReplacePolicy: Retain
    Properties:
      DBClusterIdentifier: FooBar
      Engine: !Ref Engine
      StorageEncrypted: true
      ## XXX Other properties removed for brevity

  DbWriterInstance:
    Type: AWS::RDS::DBInstance
    DeletionPolicy: Snapshot
    UpdateReplacePolicy: Retain
    Properties:
      DBClusterIdentifier: !Ref DbCluster
      Engine: !Ref Engine  # XXX here a cfn-lint bug. Allowed parameter values are "aurora-postgresql" and "aurora-mysql"
      PubliclyAccessible: false
```
